src/ai/backend/storage/kmanila/__init__.py

```diff
@@ -0,0 +1,272 @@
+                        "password": {
+                            "user": {
+                                "domain": {"id": "default"},
+                                "name": rqst_id,
+                                "password": rqst_pwd,
+                            }
+                        },
+                    },
+                    "scope": {
+                        "project": {
+                            "domain": {"id": "default"},
+                            "name": rqst_id,
+                        }
+                    },
+                }
+            }
+            headers = {"Content-Type": "application/json", "Accept": "application/json"}
+            async with sess.post(
+                self.api_base_url / "d3/identity/auth/tokens", headers=headers, json=request_body
+            ) as resp:
+                token = resp.headers.get("X-Subject-Token")
+                if token is None:
+                    status_code = resp.status
+                    raise ExecutionError(f"Token not found. {status_code = }, {rqst_id = }")
+                if self.default_project_id is None:
+                    data: dict[str, Any] = await resp.json()
+                    project_id = data["token"]["project"]["id"]
+                else:
+                    project_id = self.default_project_id
+                return {"project_id": project_id, "auth_token": token}
+
+    async def get_user_volume_id(self, quota_scope_id: QuotaScopeID) -> str | None:
+        user_vol = await self.etcd.get_prefix(
+            f"{QUOTA_VOLUME_ID_MAP_KEY}/{quota_scope_id.scope_id}"
+        )
+        if not user_vol:
+            return None
+        return str(user_vol["volume_id"])
+
+    async def put_user_volume_id(self, quota_scope_id: QuotaScopeID, data: QuotaVolumeMap) -> None:
+        await self.etcd.put_prefix(f"{QUOTA_VOLUME_ID_MAP_KEY}/{quota_scope_id.scope_id}", data)  # type: ignore[arg-type]
+
+    async def get_volume_id(
+        self,
+        session: aiohttp.ClientSession,
+        quota_scope_id: QuotaScopeID,
+        auth_info: dict[str, Any],
+        *,
+        do_hard_check: bool = True,
+    ) -> str | None:
+        if (vol_id := await self.get_user_volume_id(quota_scope_id)) is not None:
+            if not do_hard_check:
+                return vol_id
+        project_id = auth_info["project_id"]
+        async with session.get(f"/d3/adm/manila/v2/{project_id}/shares/{vol_id}") as resp:
+            if resp.status == 200:
+                volume_info = (await resp.json())["share"]
+                return volume_info["id"]
+            elif resp.status // 100 == 5:
+                raise ExternalError("Cannot get data from API server")
+        return None
+
+    async def _create_volume(
+        self,
+        session: aiohttp.ClientSession,
+        quota_scope_id: QuotaScopeID,
+        options: QuotaConfig | None,
+        auth_info: dict[str, Any],
+        *,
+        name: str,
+    ) -> str:
```

Comment:
If you were to return value of intrinsic type the callee can be confused by what this function actually returns. Please write down the comment about what this actually does. This comment also applies to `_create_access_control()`.